src/coreclr/nativeaot/Runtime/wasm/wasi_pthread.c

```diff
@@ -0,0 +1,102 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// This file exists to get access to the otherwise internal values of the __pthread.
+
+#include "wasi.h"
+#include <pthread.h>
+
+// copied from internal/pthread_impl.h
```

Comment:
Where was this copied from exactly - what is the license on the file that this copied from?

Reply:
https://github.com/WebAssembly/wasi-libc/blob/main/libc-top-half/musl/src/internal/pthread_impl.h

There's no header on that file, but https://github.com/WebAssembly/wasi-libc/tree/1dfe5c302d1c5ab621f7abf04620fae92700fd22 has various license files.  As this is `libc-top-half/musl` it looks like it is MIT

https://github.com/WebAssembly/wasi-libc/blob/1dfe5c302d1c5ab621f7abf04620fae92700fd22/LICENSE#LL12C7-L12C72

and

https://github.com/WebAssembly/wasi-libc/blob/1dfe5c302d1c5ab621f7abf04620fae92700fd22/libc-top-half/musl/COPYRIGHT#LL1C1-L1C1